src/test/crux-wallet-client.ts

```diff
@@ -0,0 +1,152 @@
+        });
+
+        it('Invalid ID', async function() {
+            const promise = this.cc.resolveCurrencyAddressForCruxID('lolwamax', 'bitcoin');
+            await expect(promise).to.be.eventually.rejected.with.property('errorCode', PackageErrorCode.CruxIdInvalidStructure);
+        });
+        it('Wallet doesnt have asset id mapped', async function() {
+            const promise = this.cc.resolveCurrencyAddressForCruxID(testCruxUser.cruxID.toString(), 'foo');
+            await expect(promise).to.be.eventually.rejected.with.property('errorCode', PackageErrorCode.AssetIDNotAvailable);
+        });
+        it('User doesnt have a currency address', async function() {
+            const promise = this.cc.resolveCurrencyAddressForCruxID(testCruxUser.cruxID.toString(), 'ethereum');
+            await expect(promise).to.be.eventually.rejected.with.property('errorCode', PackageErrorCode.AddressNotAvailable);
+        });
+        it('ID is case insensitive', async function() {
+            const address = await this.cc.resolveCurrencyAddressForCruxID('Foo123@testwallet.crux', 'bitcoin');
+            await expect(address).to.have.property('addressHash').equals('foobtcaddress');
+        });
+
+    });
+
+    describe('ID Availability check', function() {
+        beforeEach(function() {
+            this.cc = new CruxWalletClient({
+                walletClientName: 'somewallet'
+            });
+        });
+        it('Available ID check', async function() {
+            expect(await this.cc.isCruxIDAvailable('random123')).equals(true);
+        });
+        it('Unavailable ID check', async function() {
+            expect(await this.cc.isCruxIDAvailable(testCruxUser2.cruxID.components.subdomain)).equals(false);
+        });
+    });
+
+    it('New ID Registration works properly', async function() {
+        let cc = new CruxWalletClient({
+            walletClientName: 'somewallet',
+            privateKey: testPvtKey
+        });
+        const initIdState = await cc.getCruxIDState();
+        expect(initIdState.cruxID).to.equals(null);
+
+        await cc.registerCruxID('newtestuser');
+        const idState = await cc.getCruxIDState();
+        expect(idState.cruxID).equals('newtestuser@somewallet.crux');
+        expect(idState.status.status).equals(SubdomainRegistrationStatus.PENDING);
+    });
+
+    describe('Client tests with private key of existing user', async function() {
+
+        beforeEach(function() {
+            this.cc = new CruxWalletClient({
+                walletClientName: 'somewallet',
+                privateKey: testPvtKey2
+            });
+        });
+
+        it('User is recovered properly from private key', async function() {
+            const idState: ICruxIDState = await this.cc.getCruxIDState();
+            expect(idState.cruxID!.toString()).equals(testCruxUser2.cruxID.toString());
+        });
+        it('Cannot register because user is already registered', async function() {
```

Comment:
Does it cover the "PENDING"/"PUBLISHED (still waiting for confirmation on bitcoin for BNS resolution)"  state in the registrar as well?

Reply:
https://github.com/cruxprotocol/js-sdk/pull/152/commits/fcd1c28981576ed8be56601fb71480e4d9dcbae6